pkg/operator/k8sutil/network_test.go

```diff
@@ -29,18 +29,45 @@ func TestNetwork_ApplyMultusShort(t *testing.T) {
 	net := rookv1.NetworkSpec{
 		Provider: "multus",
 		Selectors: map[string]string{
-			"server": "macvlan@net1",
-			"broker": "macvlan@net2",
+			publicNetworkSelectorKeyName:  "macvlan@net1",
+			clusterNetworkSelectorKeyName: "macvlan@net2",
 		},
 	}
 
-	objMeta := metav1.ObjectMeta{}
-	err := ApplyMultus(net, &objMeta)
-	assert.NoError(t, err)
+	tests := []struct {
+		name    string
+		labels  map[string]string
+		want    []string
+		notWant []string
+	}{
+		{
+			name:    "for a non osd pod",
+			want:    []string{"macvlan@net1"},
+			notWant: []string{"macvlan@net2"},
+		},
+		{
+			name:   "for an osd pod",
+			want:   []string{"macvlan@net1", "macvlan@net2"},
+			labels: map[string]string{"app": "rook-ceph-osd"},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			objMeta := metav1.ObjectMeta{}
+			objMeta.Labels = test.labels
+			err := ApplyMultus(net, &objMeta)
+			assert.NoError(t, err)
+
+			assert.Contains(t, objMeta.Annotations, "k8s.v1.cni.cncf.io/networks")
+			for _, want := range test.want {
+				assert.Contains(t, objMeta.Annotations["k8s.v1.cni.cncf.io/networks"], want)
+			}
+			for _, notWant := range test.notWant {
+				assert.NotContains(t, objMeta.Annotations["k8s.v1.cni.cncf.io/networks"], notWant)
+			}
```

Comment:
It's better to verify the output is **exactly** what is desired. This code as written does not protect future contributions from adding undesirable content to the annotations.

This is a more exact test.
```go
assert.ElementsMatch(t, objectMeta.Annotations["k8s.v1.cni.cncf.io/networks"], want)
```

Reply:
we are comparing strings i.e., `objectMeta.Annotations["k8s.v1.cni.cncf.io/networks"]` and `want`
can we use the following
```go
assert.Equal(t, objectMeta.Annotations["k8s.v1.cni.cncf.io/networks"], want)
```

Reply:
or
```go
assert.Exactly(t, objMeta.Annotations, want)
```

Reply:
done